src/board.cpp

```
@@ -0,0 +1,137 @@
+#include "board.hpp"
+
+// Reads in FEN string and converts it to 2D array
+// TODO: add castling availability, en passant target square, halfmove clock, and fullmove number
+Board::Board(std::string FEN)
+{
+    using enum ChessPieceType;
+    int row = 8;
+    int col = 0;
+    ChessPieceType piece;
+
+    for (char c : FEN) {
+        // '/' denotes end of row, subtract from row and reset col
+        if (c == '/') {
+            row--;
+            col = 0;
+        }
+        // c is digit (for now) only when there are empty tiles
+        else if (isdigit(c)) {
+            int clearTiles = c - '0';
+            for (int i = 0; i < clearTiles; i++) {
+                boardState[row][col] = none;
+                col++;
+                }
+        }
+        // Active color
+        else if (c == 'b' || c == 'w') {
+            if (c == 'b')
+                m_color = 1;
+            else
+                m_color = -1;
+        }
+        // piece case
+        else {
+            switch (c) {
+            case 'p': piece = black_pawn; col++; break;
+            case 'r': piece = black_rook; col++; break;
+            case 'n': piece = black_knight; col++; break;
+            case 'b': piece = black_bishop; col++; break;
+            case 'q': piece = black_queen; col++; break;
+            case 'k': piece = black_king; col++; break;
+            case 'P': piece = white_pawn; col++; break;
+            case 'R': piece = white_rook; col++; break;
+            case 'N': piece = white_knight; col++; break;
+            case 'B': piece = white_bishop; col++; break;
+            case 'Q': piece = white_queen; col++; break;
+            case 'K': piece = white_king; col++; break;
+            default: break;
```

This is undefined if you somehow get a case where `piece` isnt initialized yet. Should probably throw an error since this case shouldn't happen. use `CORE_ASSERT(false, error_msg)` from my code

---

This function as a whole will probably be used far less than the rest of the class so I wouldn't commit too much time towards it if the parsing gets too annoying